actions/ulog_ingestion/src/ulog_ingestion/utils.py

```diff
@@ -0,0 +1,324 @@
+    """
+    Converts a field value to its corresponding JSON type.
+
+    Args:
+    - field_type: The type of the field as a string.
+    - value: The value to be converted.
+
+    Returns:
+    - The converted value in its appropriate JSON type.
+    """
+    json_type = TYPE_MAPPING.get(field_type, "string")
+    if json_type == "integer":
+        return int(value)
+    elif json_type == "number":
+        return float(value)
+    elif json_type == "boolean":
+        return bool(int(value))
+    elif json_type == "string":
+        return str(value)
+    else:
+        return value
+
+
+def create_message_path_records(topic: Any, field_data: Any) -> None:
+    """
+    Creates message path records for a given topic.
+
+    Args:
+    - topic: The topic object to which the message paths will be added.
+    - field_data: A list of field data objects containing the message definition.
+    """
+    array_list = list()
+    for field in field_data:
+        # For now only allow these types. TODO: Add support for nested types
+        if "[" in field.field_name:
+            array_name = field.field_name.split("[")[0]
+            array_field_type = f"{field.type_str}[]"
+
+            if array_name not in array_list:
+                topic.add_message_path(
+                    request=topics.AddMessagePathRequest(
+                        message_path=array_name,
+                        data_type=array_field_type,  # TBD
+                        canonical_data_type=topics.CanonicalDataType.Array,
+                    )
+                )
+
+                print(
+                    f"Adding array: {array_name}, type: {array_field_type}, canonical: {topics.CanonicalDataType.Array}"
+                )
+
+                # Add another message path for the array elements
+                sub_array_name = f"{array_name}.[*]"
+
+                canonical_data_type = TYPE_MAPPING_CANONICAL[field.type_str]
+                if field.type_str not in TYPE_MAPPING_CANONICAL.keys():
+                    canonical_data_type = topics.CanonicalDataType.Unknown
+
+                topic.add_message_path(
+                    request=topics.AddMessagePathRequest(
+                        message_path=sub_array_name,
+                        data_type=field.type_str,  # TBD
+                        canonical_data_type=canonical_data_type,
+                    )
+                )
+
+                print(
+                    f"Adding sub-field for array: {sub_array_name}, type: {field.type_str}, canonical: {TYPE_MAPPING_CANONICAL[field.type_str]}"
+                )
+
+                array_list.append(array_name)
+        else:
+            print(
+                f"Adding field: {field.field_name}, type: {field.type_str}, canonical: {TYPE_MAPPING_CANONICAL[field.type_str]}"
+            )
+
+            canonical_data_type = TYPE_MAPPING_CANONICAL[field.type_str]
+            if field.type_str not in TYPE_MAPPING_CANONICAL.keys():
```

Comment:
This statement will never run if `field.type_str` is not in `TYPE_MAPPING_CANONICAL`. Using attribute access (as on ln. 208) will raise a `KeyError`.

Reply:
Good catch - fixed it